Makefile

```diff
@@ -41,7 +41,7 @@ ifeq (, $(findstring pse, $(PROC)))
 	ERROR = "1MB huge pages support is required"
 endif
 
-TFW_CFLAGS += -mmmx -msse4.2
+TFW_CFLAGS += -march=haswell
```

Comment:
I have an issue during build on my broadwell cpu in qemu:
```
 CC [M]  /home/user/tempesta/tempesta/tls/aesni.o
/home/user/tempesta/tempesta/tls/aesni.c: In function ‘mbedtls_aesni_crypt_ecb’:
/home/user/tempesta/tempesta/tls/aesni.c:89:5: error: unknown register name ‘xmm1’ in ‘asm’
     asm( "movdqu    (%3), %%xmm0    \n\t" // load input
     ^~~
/home/user/tempesta/tempesta/tls/aesni.c:89:5: error: unknown register name ‘xmm0’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c: In function ‘mbedtls_aesni_gcm_mult’:
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm5’ in ‘asm’
     asm( "movdqu (%0), %%xmm0               \n\t" // a1:a0
     ^~~
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm4’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm3’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm2’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm1’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c:144:5: error: unknown register name ‘xmm0’ in ‘asm’
/home/user/tempesta/tempesta/tls/aesni.c: In function ‘mbedtls_aesni_inverse_key’:
/home/user/tempesta/tempesta/tls/aesni.c:250:9: error: unknown register name ‘xmm0’ in ‘asm’
         asm( "movdqu (%0), %%xmm0       \n\t"
         ^~~
/usr/src/linux-headers-4.14.0-tempesta-common/scripts/Makefile.build:333: recipe for target '/home/user/tempesta/tempesta/tls/aesni.o' failed
make[5]: *** [/home/user/tempesta/tempesta/tls/aesni.o] Error 1
/usr/src/linux-headers-4.14.0-tempesta-common/scripts/Makefile.build:592: recipe for target '/home/user/tempesta/tempesta/tls' failed
make[4]: *** [/home/user/tempesta/tempesta/tls] Error 2
make[4]: *** Waiting for unfinished jobs....
```
Here is the flags used to build the `aesni.c`:
```
   gcc-6 -Wp,-MD,/home/user/tempesta/tempesta/tls/.aesni.o.d  -nostdinc -isystem /usr/lib/gcc/x86_64-linux-gnu/6/include -I/usr/src/linux-headers-4.14.0-tempesta-common/arch/x86/include -I./arch/x86/include/generated  -I/usr/src/linux-headers-4.14.0-tempesta-common/include -I./include -I/usr/src/linux-headers-4.14.0-tempesta-common/arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I/usr/src/linux-headers-4.14.0-tempesta-common/include/uapi -I./include/generated/uapi -include /usr/src/linux-headers-4.14.0-tempesta-common/include/linux/kconfig.h  -I/home/user/tempesta/tempesta/tls -I/home/user/tempesta/tempesta/tls -D__KERNEL__ -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -Werror-implicit-function-declaration -Wno-format-security -std=gnu89 -fno-PIE -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64 -falign-jumps=1 -falign-loops=1 -mno-80387 -mno-fp-ret-in-387 -mpreferred-stack-boundary=3 -mskip-rax-setup -mtune=generic -mno-red-zone -mcmodel=kernel -funit-at-a-time -DCONFIG_X86_X32_ABI -DCONFIG_AS_CFI=1 -DCONFIG_AS_CFI_SIGNAL_FRAME=1 -DCONFIG_AS_CFI_SECTIONS=1 -DCONFIG_AS_FXSAVEQ=1 -DCONFIG_AS_SSSE3=1 -DCONFIG_AS_CRC32=1 -DCONFIG_AS_AVX=1 -DCONFIG_AS_AVX2=1 -DCONFIG_AS_AVX512=1 -DCONFIG_AS_SHA1_NI=1 -DCONFIG_AS_SHA256_NI=1 -pipe -Wno-sign-compare -fno-asynchronous-unwind-tables -mindirect-branch=thunk-extern -mindirect-branch-register -DRETPOLINE -fno-delete-null-pointer-checks -Wno-frame-address -O2 --param=allow-store-data-races=0 -DCC_HAVE_ASM_GOTO -Wframe-larger-than=2048 -fstack-protector-strong -Wno-unused-but-set-variable -Wno-unused-const-variable -fno-var-tracking-assignments -g -pg -mfentry -DCC_USING_FENTRY -Wdeclaration-after-statement -Wno-pointer-sign -fno-strict-overflow -fno-merge-all-constants -fmerge-constants -fno-stack-check -fconserve-stack -Werror=implicit-int -Werror=strict-prototypes -Werror=date-time -Werror=incompatible-pointer-types -Werror=designated-init -Werror -mpreferred-stack-boundary=4 -DAVX2=1 -march=broadwell  -I/home/user/tempesta/tempesta/tls  -I/home/user/tempesta/tempesta/tls/dummy_headers -include ttls.h  -DMODULE  -DKBUILD_BASENAME='"aesni"'  -DKBUILD_MODNAME='"tempesta_tls"' -c -o /home/user/tempesta/tempesta/tls/.tmp_aesni.o /home/user/tempesta/tempesta/tls/aesni.c
```
I am able to compile using old `TFW_CFLAGS`.

Reply:
The same issue reproduced on a the 8c server. But I don't see any review from buildbot. @vladtcvs  something is wrong with it?